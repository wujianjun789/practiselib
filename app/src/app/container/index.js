/**
 * Created by a on 2017/6/29.
 */
import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux'
import PropTypes from 'prop-types';
import '../../../public/styles/app.less';
import Card from './Card';
import UserCenter from '../../common/containers/UserCenter';
import Overlayer from '../../common/containers/Overlayer';
import {loginHandler} from '../../api/login'
import {getModule} from '../action/index'
/**
 * @param {String} title required
 * @param {String} name  required
 * @param {Array}  items required
 */
export class App extends Component{
    constructor(props){
        super(props);
        this.loginFail = this.loginFail.bind(this);
        this.loginSuccess = this.loginSuccess.bind(this);
    }

    componentWillMount(){
        console.log(localStorage)
        /*if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            
            this.props.router.push('/login')
            
        }else{

            this.props.actions.loginHandler(sessionStorage.username, sessionStorage.password, this.loginSuccess, this.loginFail);

        }*/
        // if(this.props.userCenter.islogin!=1){
        //     this.props.router.push('/login')
        // }
        const {actions} = this.props;
        actions && actions.getModule();
    }

    componentDidMount() {
    }

    componentWillReceiveProps(nextProps) {



    }

    loginFail(){
        this.setState({style:{visibility: 'visible'}})
    }

    loginSuccess(){
        this.props.router.push('/')
    }

    render(){
        const {title="StarRiver", name="智慧路灯管理系统", items} = this.props;
        return (
            <div className="app">
                <div className="header clearfix">
                    <div className="header-left clearfix">
                        <span className="icon icon-logo"></span>
                        <span className="tit">{title}</span>
                    </div>
                    <UserCenter router={this.props.router}/>
                    <div className="header-right clearfix">
                        <span className="name">{name}</span>
                    </div>
                </div>
                <div className="cont">
                    <ul className="clearfix">
                        {items.map((item, index)=><li key={item.key}><Card _key={item.key} title={item.title} link={item.link}/></li>)}
                    </ul>
                </div>
                <Overlayer />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    let app = state.app;
    return {
        title: app.title,
        name: app.name,
        items: app.items,
        userCenter:state.userCenter
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        actions: bindActionCreators({
            getModule: getModule,
            loginHandler
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
    title: PropTypes.string,
    name: PropTypes.string,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            key: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
            link: PropTypes.string.isRequired
        })
    ).isRequired
}