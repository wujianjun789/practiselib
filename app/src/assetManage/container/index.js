/**
 * Created by a on 2017/7/17.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {injectIntl} from 'react-intl';

import '../../../public/styles/assetManage-model.less';
import '../../../public/styles/assetManage-statistics.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '../../data/assetModels';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';

import {sideBarToggled} from '../action/index';
import {treeViewNavigator} from '../../common/util/index';
class AssetManageIndex extends Component {
  constructor(props) {
    super(props);

    this.state = {
      pathname: ""
    }

    this.initModel = this.initModel.bind(this);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);

    this.getMathNode = this.getMathNode.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    this.initModel();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !==  this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }
  componentDidUpdate(){
    // console.log(this.props);
    // const {location} = this.props;
    // if(location.pathname !== this.state.pathname){
    //   this.setState({pathname: location.pathname}, ()=>{
    //     this.initModel();
    //   });
    // }
  }
  initModel(){
    getModelData(() => {this.mounted && this.initTreeData();});
  }

  initTreeData() {
    TreeData.map(tree=>{
      return Object.assign({}, tree, {name:this.props.intl.formatMessage({id:tree.name})});
    })
    this.props.actions.treeViewInit(TreeData);
    treeViewNavigator(TreeData, this.props.router);

    const node = this.getMathNode(TreeData, this.props.location.pathname);
    if(node){
      this.onToggle(node);
    }
  }

  getMathNode(data, pathname){
    for(let i=0;i<data.length;i++){
      if(!data[i].children && data[i].link.indexOf(pathname) > -1){
        return data[i];
      }else{
        if(data[i].children){
          const node = this.getMathNode(data[i].children, pathname);
          if(node){
            return node;
          }
        }

      }
    }

    return null;
  }

  onToggle(node) {
    const {actions} = this.props;
    actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }

    return <div className={'container ' + 'asset-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName="app.asset.manage" router={this.props.router}/>
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById,
      sideBarToggled: sideBarToggled,
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(AssetManageIndex));