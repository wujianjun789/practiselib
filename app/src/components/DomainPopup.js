import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import Panel from './Panel';
import PanelFooter from './PanelFooter';
/**
 * Domain component
 * @param {String}      title       'domain title,
 * @param {Object}      data        'domain data,  example:{domainName:"域名", lng:35, lat: 65, prevDomain: "上级域"}　'
 * @param {Object}      domainList  
 * @param {String}      domainList.titileKey 
 * @param {String}      domainList.vlaueKey  
 * @param {Array}       domainList.options     'example:[{id:0, value:"上级域1", title:"上级域1"}]'
 * @param {Func}        onConfirm   'panel save button handler'
 * @param {Func}        onCancel    'panel close button handler'
 * 
 */

import MapView from './MapView'

import {latlngValid} from '../util/index'
export default class DomainPopup extends PureComponent {
    constructor(props) {
        super(props);
        const {domainId, domainName, lat, lng, prevDomain} = this.props.data;
        let {options} = this.props.domainList;
        let curDomain = options && options.length ? options[0]:null
        this.state = {
            domainId: domainId,
            domainName: domainName,
            lng: lng,
            lat: lat,
            prevDomain: prevDomain?prevDomain:(curDomain?curDomain.id:"")
        }
        this.onConfirm = this.onConfirm.bind(this);
        this.onCancel = this.onCancel.bind(this);
        this.onChange = this.onChange.bind(this);
    }

    onConfirm() {
        this.props.onConfirm(this.state);
    }

    onCancel() {
        this.props.onCancel();
    }

    onChange(e) {
        let id = e.target.id;

        if(id=="prevDomain"){
            const {options} = this.props.domainList;
            let curIndex = e.target.selectedIndex;
            this.setState({[id]:options[curIndex].id});
            return
        }
        let value = e.target.value;
        let newValue='';
        if(id == "lat" || id == "lng"){
            for(let i=0;i<value.length;i++)
            {
                let s = value.slice(i, i+1);
                if(latlngValid(s)){
                    newValue += s;
                }
            }

            newValue = Number(newValue);

        }else{
            newValue = value;
        }
        this.setState({[id]: newValue});
    }

    render() {
         let {domainId, domainName, lng, lat, prevDomain} = this.state;
         let {titleKey, valueKey, options} = this.props.domainList;
         let footer = <PanelFooter funcNames={['onCancel','onConfirm']} btnTitles={['取消','保存']} 
            btnClassName={['btn-default', 'btn-primary']} 
            btnDisabled={[false, false]} onCancel={this.onCancel} onConfirm={this.onConfirm}/>;
        let curDomain = null;
        for(let key in options){
            if(options[key].id == prevDomain){
                curDomain = options[key];
                break;
            }
        }

        return <div className="domain-popup">
            <Panel title={this.props.title} closeBtn={true} closeClick={this.onCancel} >
                <div className="row">
                    <div className="col-sm-6 popup-left">
                        <div className="form-group row">
                            <label className="col-sm-3 control-label" htmlFor="domainName">域名称：</label>
                            <div className="col-sm-9">
                                <input type="text" className="form-control" id="domainName" placeholder="输入域名称" value={domainName} onChange={this.onChange} />
                            </div>
                        </div>
                        <div className="form-group row">
                            <label className="col-sm-3 control-label " htmlFor="lng">经度：</label>
                            <div className="col-sm-9">
                                <input type="email" className="form-control" id="lng" placeholder="输入GPS坐标" value={lng}
                                        onChange={this.onChange}/>
                            </div>
                        </div> 
                        <div className="form-group row">   
                            <label className="col-sm-3 control-label" htmlFor="lat">纬度：</label>
                            <div className="col-sm-9">
                                <input type="email" className="form-control" id="lat" placeholder="输入GPS坐标" value={lat} onChange={this.onChange}/>
                            </div>
                        </div>
                        <div className="form-group row">
                            <label className="col-sm-3 control-label" htmlFor="prevDomain">上级域：</label>
                            <div className="col-sm-9">
                                <select className="form-control" id="prevDomain" placeholder="选择上级域" value={curDomain.name} onChange={this.onChange}>
                                    {
                                        options.map(item => <option key={item.id} value={item[valueKey]}>{item[titleKey]}</option>)
                                    }
                                </select>
                            </div>
                        </div>
                        {footer}
                    </div>
                    <div className="col-sm-6 popup-map">
                        {
                            lng && lat ? <MapView option={{mapZoom:false}} mapData={{id:"domainPopup",  latlng:{lng:lng, lat:lat},
                        position:[{"device_id":domainId, "device_type":"DEVICE",lng:lng, lat:lat}], data:[{id:domainId, name:domainName}]}}/>
                        :''}
                    </div>
                </div>
            </Panel>
        </div>
    }
}

DomainPopup.propTypes = {
    title: PropTypes.string.isRequired,
    data: PropTypes.shape({
        domainName: PropTypes.string.isRequired,
        lat: PropTypes.number.isRequired,
        lng: PropTypes.number.isRequired,
        prevDomain: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number
        ]).isRequired
    }).isRequired,
    domainList: PropTypes.shape({
        titleKey: PropTypes.string.isRequired,
        valueKey: PropTypes.string.isRequired,
        options: PropTypes.array.isRequired
    }).isRequired,
    onConfirm: PropTypes.func.isRequired,
    onCancel: PropTypes.func.isRequired
}