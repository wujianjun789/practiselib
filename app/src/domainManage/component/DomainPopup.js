import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import Panel from './../../components/Panel';
import PanelFooter from './../../components/PanelFooter';

import NotifyPopup from '../../common/containers/NotifyPopup'
/**
 * Domain component
 * @param {String}      title       'domain title,
 * @param {Object}      data        'domain data,  example:{domainName:"域名", lng:35, lat: 65, prevDomain: "上级域"}　'
 * @param {Object}      domainList  
 * @param {String}      domainList.titileKey 
 * @param {String}      domainList.vlaueKey  
 * @param {Array}       domainList.options     'example:[{id:0, value:"上级域1", title:"上级域1"}]'
 * @param {Func}        onConfirm   'panel save button handler'
 * @param {Func}        onCancel    'panel close button handler'
 * 
 */

import MapView from './../../components/MapView'

import {FormattedMessage,injectIntl} from 'react-intl';
import { intlFormat } from '../../util/index';

import {Name2Valid, latlngValid, lngValid, latValid} from '../../util/index'
export default class DomainPopup extends PureComponent {
    constructor(props) {
        super(props);
        const {domainId, domainName, lat, lng, prevDomain} = this.props.data;
        let {options} = this.props.domainList;
        let curDomain = options && options.length ? options[0]:null
        this.state = {
            domainId: domainId,
            domainName: domainName,
            zoom: 16,
            lng: lng,
            lat: lat,
            prevDomain: prevDomain?prevDomain:(curDomain?curDomain.id:""),
            prompt:{
                // domainName:Boolean(domainName),
                // lng: Boolean(lng),
                // lat: Boolean(lat)
                domainName:false,
                lng: false,
                lat: false
            },
            addFirstTime:this.props.idAdd//如果是add并且是第一次加载页面，不提示非法输入
        }
        this.onConfirm = this.onConfirm.bind(this);
        this.onCancel = this.onCancel.bind(this);
        this.onChange = this.onChange.bind(this);
        this.mapDragend = this.mapDragend.bind(this);
    }

    componentWillMount(){
        this.mounted = false;
    }

    componentWillUnmout(){
        this.mounted = true;
    }

    onConfirm() {
        this.props.onConfirm && this.props.onConfirm(this.state);
    }

    onCancel() {
        this.props.onCancel();
    }

    onChange(e) {
        let id = e.target.id;

        if(id=="prevDomain"){
            const {options} = this.props.domainList;
            let curIndex = e.target.selectedIndex;
            this.setState({[id]:options[curIndex].id});
            return
        }
        let value = e.target.value;
        let newValue='';
        let prompt = false
        if(id == "lat" ){
            newValue = value;
            if(!latlngValid(newValue) || !latValid(newValue)){
                prompt = true;
            }
        }else if(id== "lng"){
            newValue = value;
            if(!latlngValid(newValue) || !lngValid(newValue)){
                prompt = true
            }
        }else if(id == "domainName"){
            newValue = value;
            prompt = !Name2Valid(newValue);
        }else{
            newValue = value;
        }

        this.setState({[id]: newValue, prompt:Object.assign({}, this.state.prompt, {[id]:prompt})});
    }

    mapDragend(data){
        // console.log(data.bounds);
        let obj = {};
        let prompt = {};
        if(data.zoom){
            obj = Object.assign({}, {zoom:data.zoom});
        }

        if(data.latlng){
            obj = Object.assign({}, obj, {lng:data.latlng.lng}, {lat:data.latlng.lat})
            prompt = Object.assign({}, {lng:false}, {lat:false});
        }
        this.setState(Object.assign({}, this.state, obj, {prompt: Object.assign({}, this.state.prompt, prompt)}));
    }

    render() {
         let {domainId, domainName, zoom, lng, lat, prevDomain, prompt} = this.state;
         let {titleKey, valueKey, options} = this.props.domainList;

        let valid = prompt.domainName || prompt.lng || prompt.lat || !domainName || !lat || !lng;

         let footer = <PanelFooter funcNames={['onCancel','onConfirm']} btnTitles={['button.cancel','button.confirm']}
            btnClassName={['btn-default', 'btn-primary']} 
            btnDisabled={[false, valid]} onCancel={this.onCancel} onConfirm={this.onConfirm}/>;
        let curDomain = null;
        for(let key in options){
            if(options[key].id == prevDomain){
                curDomain = options[key];
                break;
            }
        }

        return <div className="domain-popup">
            <Panel title={this.props.title} closeBtn={true} closeClick={this.onCancel} >
                <div className="row">
                    <div className="col-sm-6 col-xs-6 popup-left">
                        <div className="form-group row">
                            <label className="fixed-width-left control-label" htmlFor="domainName"><FormattedMessage id='domain.name'/>：</label>
                            <div className="fixed-width-right">
                                <input type="text" className={ "form-control " } id="domainName" placeholder={intlFormat({en:'please input the name',zh:'输入域名称'})} maxLength="16" value={domainName}
                                       onChange={this.onChange}/>
                                <span className={prompt.domainName?"prompt ":"prompt hidden"}>{intlFormat({en:'illegal',zh:'仅能使用字母、数字或下划线'})}</span>
                            </div>
                        </div>
                        <div className="form-group row">
                            <label className="fixed-width-left control-label " htmlFor="lng"><FormattedMessage id='domain.lng'/>：</label>
                            <div className="fixed-width-right">
                                <input type="email" className={ "form-control " } id="lng" placeholder={intlFormat({en:'please input the GPS',zh:'输入GPS坐标'})} value={lng}
                                        onChange={this.onChange}/>
                                <span className={prompt.lng?"prompt ":"prompt hidden"}>{intlFormat({en:'illegal',zh:'经度数不合法'})}</span>
                            </div>
                        </div> 
                        <div className="form-group row">   
                            <label className="fixed-width-left control-label" htmlFor="lat"><FormattedMessage id='domain.lat'/>：</label>
                            <div className="fixed-width-right">
                                <input type="email" className={ "form-control " } id="lat" placeholder={intlFormat({en:'please input the GPS',zh:'输入GPS坐标'})} value={lat} onChange={this.onChange}/>
                                <span className={prompt.lat?"prompt ":"prompt hidden"}>{intlFormat({en:'illegal',zh:'纬度数不合法'})}</span>
                            </div>
                        </div>
                        <div className="form-group row">
                            <label className="fixed-width-left control-label" htmlFor="prevDomain"><FormattedMessage id='domain.superior'/>：</label>
                            <div className="fixed-width-right">
                                <select className="form-control" id="prevDomain" disabled={this.props.id=="updateDomain"?true:false} placeholder={intlFormat({en:'please input the superior domain',zh:'输入GPS坐标'})} value={curDomain?curDomain.name:intlFormat({en:'null',zh:'无'})} onChange={this.onChange}>
                                    {
                                        options.map(item => <option key={item.id} value={item[valueKey]}>{item[titleKey]}</option>)
                                    }
                                </select>
                            </div>
                        </div>
                        {footer}
                    </div>
                    <div className="col-sm-6 col-xs-6 popup-map">

                            <MapView mapIcon={true} option={{zoom:zoom, mapZoom:false, markerDraggable:true}} mapData={{id:"domainPopup",  latlng:{lng:lng, lat:lat},
                        position:[/*{"device_id":domainId, "device_type":"DEVICE",lng:lng, lat:lat}*/], data:[{id:domainId, name:domainName}]}}
                                     mapCallFun={{mapDragendHandler:this.mapDragend}} markerCallFun={{markerDragendHandler:this.mapDragend}}/>
                    </div>
                </div>
                <NotifyPopup />
            </Panel>
        </div>
    }
}

DomainPopup.propTypes = {
    title: PropTypes.string.isRequired,
    data: PropTypes.shape({
        domainName: PropTypes.string.isRequired,
        // lat: PropTypes.number.isRequired,
        // lng: PropTypes.number.isRequired,
        prevDomain: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number
        ]).isRequired
    }).isRequired,
    domainList: PropTypes.shape({
        titleKey: PropTypes.string.isRequired,
        valueKey: PropTypes.string.isRequired,
        options: PropTypes.array.isRequired
    }).isRequired,
    onConfirm: PropTypes.func.isRequired,
    onCancel: PropTypes.func.isRequired
}