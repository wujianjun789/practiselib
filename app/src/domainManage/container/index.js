/**
 * Created by a on 2017/7/17.
 */
import React,{Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import HeadBar from '../../components/HeadBar'
import SideBar from '../../components/SideBar'
import Overlayer from '../../common/containers/Overlayer'

import {TreeData} from '.././../data/domainModel'
import {treeViewInit} from '../../common/actions/treeView'
import {sideBarToggled} from '../action/index'
class DomainManageIndex extends Component{
    constructor(props){
        super(props);
        this.initTreeData = this.initTreeData.bind(this);
        this.onToggle = this.onToggle.bind(this);
    }

    componentWillMount(){
        if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            this.props.router.push('/login') 
        }else{}
        this.initTreeData();
    }

    componentWillUnmount(){

    }

    componentDidMount(){
        console.log(this.props.userCenter); 
    }

    initTreeData(){
        this.props.actions.treeViewInit(TreeData);
    }

    onToggle(node){
        this.props.actions.sideBarToggled(node);
    }

    render(){
        let path = "";
        const {children} = this.props;
        if(children){
            path = children.props.route.path;
        }
        return <div className={"container "+"domain-"+path}>
            <HeadBar moduleName={/*path=="domainEdit"?*/"域管理"/*:"地图预览"*/} router={this.props.router}/>
            <SideBar onToggle={this.onToggle}/>
            {this.props.children}
            <Overlayer />
        </div>
    }
}

function mapStateToProps(state) {
    return {
        userCenter:state.userCenter
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            treeViewInit: treeViewInit,
            sideBarToggled: sideBarToggled
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DomainManageIndex);