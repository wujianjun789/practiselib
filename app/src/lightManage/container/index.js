/**
 * Created by a on 2017/8/24.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import '../../../public/styles/lightManage-map.less';
import '../../../public/styles/lightManage-control.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '.././../data/lightModel';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';
// import {sideBarToggled} from '../action/index'
class lightManageIndex extends Component {
  constructor(props) {
    super(props);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    getModelData(null, () => {
      this.mounted && this.initTreeData();
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidMount() {
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !== this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }

  initTreeData() {
    this.props.actions.treeViewInit(TreeData);
  }

  onToggle(node) {
    // console.log(node);
    // this.props.actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }
    return <div className={'container light-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName={'app.light'} router={this.props.router}/>
                    
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById
      // sideBarToggled: sideBarToggled
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(lightManageIndex);