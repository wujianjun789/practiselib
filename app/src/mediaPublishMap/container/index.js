/**
 * Created by a on 2017/10/19.
 */
import React,{Component} from 'react';

import Content from '../../components/Content';
import MapView from '../../components/MapView';
export default class MediaPublishMap extends Component{
    constructor(props){
        super(props);
        this.state = {
            map:{
                id: "mediaPublishMap",
                latlng: []
            }
        };
    }

    render(){
        const {map} = this.state;
        return <Content>
            <MapView mapData={map}></MapView>
        </Content>
    }
}