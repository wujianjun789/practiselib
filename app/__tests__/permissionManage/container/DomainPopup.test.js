import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {DomainPopup} from '../../../src/permissionManage/container/DomainPopup';
import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl-redux';
import configureStore from '../../../src/store/configureStore';
import Immutable from 'immutable';

const store = configureStore();
describe('<DomainPopup',()=>{

    const className = 'user-domain-edit-popup';
    const title = '用户域管理';
    
    it('render normal',()=>{
        const onClick = jest.fn();
        const root = shallow(<Provider store={store}>
            <IntlProvider>
                <DomainPopup className={className} title={title} onClick={onClick}/>
            </IntlProvider>
        </Provider>);
        const cmp = root.find('DomainPopup')
        const container = cmp.find(`.${className}`);
        expect(container.length).toBe(1);

        expect(cmp.length).toBe(1);
        expect(cmp.prop('title')).toBe(title);
        
        const domainPer = cmp.find('.domain-per');
        expect(domainPer.length).toBe(1);

        const domainContent = cmp.find('.domain-content');
        expect(domainContent.length).toBe(1);

        const domainTree = cmp.find('.domain-tree');
        expect(domainTree.length).toBe(1);

        const domainList = cmp.find('.domain-list');
        expect(domainList.length).toBe(1);
        
        const searchText = domainTree.find('SearchText');
        expect(searchText.length).toBe(1);

        const deleteSpan = domainList.find('icon-table-delete');
        deleteSpan[0]&&deleteSpan[0].simulate('click');
        expect(onClick).toHaveBeenCalledTimes(deleteSpan[0]?1:0);

    })

    it('snapshot',() => {
        const cmp = renderer.create(<Provider store={store}>
            <IntlProvider>
                <DomainPopup />
            </IntlProvider>
        </Provider>);
        const tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    })
})