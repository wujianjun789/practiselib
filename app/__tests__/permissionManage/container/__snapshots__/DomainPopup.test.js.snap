exports[`<DomainPopup snapshot 1`] = `
<div
  className="modal-content ">
  <div
    className="modal-header">
    <button
      aria-label="Close"
      className="close"
      data-dismiss="modal"
      onClick={[Function]}
      type="button">
      <span
        aria-hidden="true">
        ×
      </span>
    </button>
    <h4
      className="modal-title">
      
    </h4>
  </div>
  <div
    className="modal-body">
    <div
      className="form-group row domain-per">
      <label
        className="col-sm-2 control-label">
        域权限:
      </label>
      <div
        className="col-sm-10">
        <div
          className="col-sm-6 domain-add">
          <div
            className="domain-tree">
            <div
              className="searchText search"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}>
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="输入域名称"
                type="search"
                value="" />
              <ul
                className="" />
              <span
                className="glyphicon glyphicon-search"
                onClick={[Function]} />
            </div>
          </div>
        </div>
        <div
          className="col-sm-6 domain-list">
          <ul />
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal-footer">
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      取消
    </button>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button">
      确认
    </button>
  </div>
</div>
`;
